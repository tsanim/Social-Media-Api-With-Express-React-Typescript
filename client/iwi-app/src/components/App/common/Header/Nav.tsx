import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHome, faUsers, faSignOutAlt, faImages } from '@fortawesome/free-solid-svg-icons';
import { Detector } from "react-detect-offline";
import PropTypes from 'prop-types';

import URI from '../../../../config/config';
import Dropdown from '../../../Dropdown/NotificationsDropdown';
import HeaderProps from '../../../../interfaces/Components/Header/HeaderProps.interface';

function Nav(props: HeaderProps) {
    const { imageId } = props.user;

    return (
        <nav id="main-nav">
            <ul>
                <Detector
                    // tslint:disable-next-line: jsx-no-multiline-js
                    render={({ online }: { online: boolean}) => {
                        if (online) {
                            props.switchToOnline();
                        } else {
                            props.switchToOffline();
                        }

                        return (
                            
                            <li className={`status ${online ? 'online' : 'offline'}`}/>
                        )
                    }}
                />
                <li>
                    <Link to="/MyProfile">
                        <figure className="nav-profilePic">
                            <img src={`${URI}/feed/image/${imageId}`} alt="" />
                            <figcaption>{localStorage.getItem('username')}</figcaption>
                        </figure>
                    </Link>
                </li>
                <li><Link to="/"><FontAwesomeIcon icon={faHome} /> HOME</Link></li>
                <li><Link to="/discover"><FontAwesomeIcon icon={faUsers} /> DISCOVER</Link></li>
                <li><Link to="/searchPosts"><FontAwesomeIcon icon={faImages} /> SEARCH</Link></li>
                <li> <Dropdown
                    notifications={props.notifications}
                />
                </li>
                <li><Link to="/chat"><FontAwesomeIcon icon={faUsers} /> CHAT</Link></li>
                <li><Link to="/signin" onClick={props.signoutHandler}><FontAwesomeIcon icon={faSignOutAlt} /> LOGOUT</Link></li>
            </ul>
        </nav>
    )
}

Nav.propTypes = {
    user: PropTypes.object,
    resetUserPostsHandler: PropTypes.func,
    signoutHandler: PropTypes.func,
    switchToOffline: PropTypes.func,
    switchToOnline: PropTypes.func,
}

export default Nav;