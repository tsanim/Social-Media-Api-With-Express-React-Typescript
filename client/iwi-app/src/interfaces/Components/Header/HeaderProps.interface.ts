import User from "../../User/User.interface";
import Notification from "../../Feed/Notification.interface";

export default interface HeaderProps {
    user: User;
    notifications: Notification[];
    signoutHandler: () => void;
    switchToOffline: () => void;
    switchToOnline: () => void;
}