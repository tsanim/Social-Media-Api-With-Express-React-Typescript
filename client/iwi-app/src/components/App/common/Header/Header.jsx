import React from 'react';
import Nav from './Nav';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function Header(props) {
    return (
        <header>
            <div className="logo">
                <Link to="/"><h1>iWi</h1></Link>
            </div>
            {
                localStorage.getItem('username') 
                ? <Nav
                    user={props.user}
                    signoutHandler={props.signoutHandler}
                    switchToOffline={props.switchToOffline}
                    switchToOnline={props.switchToOnline}
                    notifications={props.notifications}
                    showDropDownHandler={props.showDropDownHandler}
                />
                : null
            }
        </header>
    )
}

Header.propTypes = {
    user: PropTypes.object,
    resetUserPostsHandler: PropTypes.func,
    signoutHandler: PropTypes.func,
    switchToOffline: PropTypes.func,
    switchToOnline: PropTypes.func,
}

export default Header;