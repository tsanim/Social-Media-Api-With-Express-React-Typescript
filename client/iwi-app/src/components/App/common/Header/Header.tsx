import React from 'react';
import Nav from './Nav';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import HeaderProps from '../../../../interfaces/Components/Header/HeaderProps.interface';

function Header(props: HeaderProps) {
    return (
        <header>
            <div className="logo">
                <Link to="/"><h1>iWi</h1></Link>
            </div>
            {
                localStorage.getItem('username') 
                ? <Nav
                    user={props.user}
                    signoutHandler={props.signoutHandler}
                    switchToOffline={props.switchToOffline}
                    switchToOnline={props.switchToOnline}
                    notifications={props.notifications}
                />
                : null
            }
        </header>
    )
}

Header.propTypes = {
    user: PropTypes.object,
    signoutHandler: PropTypes.func,
    switchToOffline: PropTypes.func,
    switchToOnline: PropTypes.func,
}

export default Header;