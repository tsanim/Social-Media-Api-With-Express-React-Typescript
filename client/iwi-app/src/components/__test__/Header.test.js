import React from 'react';
import Header from '../App/common/Header/Header';
import Nav from '../App/common/Header/Nav';

import { shallow } from 'enzyme';

function headerComponentSetup() {
    const props = {
        user: {},
        resetUserPostsHandler: jest.fn(),
        signoutHandler: jest.fn(),
        switchToOffline: jest.fn(),
        switchToOnline: jest.fn()
    }

    window.localStorage.setItem('username', 'Tsani');

    const enzymeWrapper = shallow(<Header {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('Header component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = headerComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div container with class "logo"', () => {
        const { enzymeWrapper } = headerComponentSetup();

        expect(enzymeWrapper.find('div').hasClass('logo')).toEqual(true);
    });

    it('should render Nav Component if localStorage has item username', () => {
        const { enzymeWrapper } = headerComponentSetup();

        expect(enzymeWrapper.find(Nav)).toHaveLength(1);
    });
});
