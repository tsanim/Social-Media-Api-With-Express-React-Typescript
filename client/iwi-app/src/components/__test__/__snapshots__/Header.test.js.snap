// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render initial layout 1`] = `
Array [
  <header>
    <div
      className="logo"
    >
      <Link
        to="/"
      >
        <h1>
          iWi
        </h1>
      </Link>
    </div>
    <Nav
      signoutHandler={[MockFunction]}
      switchToOffline={[MockFunction]}
      switchToOnline={[MockFunction]}
      user={Object {}}
    />
  </header>,
]
`;
